tests/scenarios/compat-addon-styles-test.ts

```diff
@@ -132,3 +132,83 @@ appScenarios
       });
     });
   });
+
+function setAddonNameAndCSS(addon: Project, letter: string) {
+  addon.pkg.name = `addon-${letter}`;
+  merge(addon.files, {
+    addon: {
+      styles: {
+        'addon.css': `.addon-${letter} {}`,
+      },
+    },
+  });
+}
+
+appScenarios
+  .map('compat-addon-styles', project => {
+    const addonA: Project = baseAddon();
+    setAddonNameAndCSS(addonA, 'a');
+
+    const addonB = baseAddon();
+    setAddonNameAndCSS(addonB, 'b');
+
+    const addonC = baseAddon();
+    setAddonNameAndCSS(addonC, 'c');
+
+    const addonD = baseAddon();
+    setAddonNameAndCSS(addonD, 'd');
+
+    project.addDependency(addonA);
+    project.addDependency(addonB);
+    addonA.addDependency(addonC);
+    project.addDependency(addonD);
+
+    project.addDependency('third-party', '1.2.3').files = {
+      'third-party.css': '.third-party {}',
+    };
+
+    merge(project.files, {
+      'ember-cli-build.js': `
+        'use strict';
+
+        const EmberApp = require('ember-cli/lib/broccoli/ember-app');
+        const { maybeEmbroider } = require('@embroider/test-setup');
+
+        module.exports = function (defaults) {
+          let app = new EmberApp(defaults, {});
+          app.import('node_modules/third-party/third-party.css');
```

Comment:
add more of these so we get coverage on the order between app.import calls as well